pkg/plugins/builtin/ssh/ssh.go

```diff
@@ -177,34 +177,30 @@ trap printResultJSON EXIT
 	}
 
 	outStr := string(output)
-	ret := map[string]interface{}{
+	metadata := map[string]interface{}{
 		"output":      outStr,
 		"exit_status": fmt.Sprint(exitStatus),
 		"exit_signal": exitSignal,
 		"exit_msg":    exitMessage,
 	}
 
-	// Unmarshal printResultJSON's output
 	lastNL := strings.LastIndexByte(outStr, '{')
-	if lastNL != -1 {
-		lastLine := outStr[lastNL:]
-		m := map[string]string{}
-		err = json.Unmarshal([]byte(lastLine), &m)
-		if err != nil {
-			return nil, nil, err
-		}
-		for k, v := range m {
-			_, exists := ret[k]
-			// do not overwrite entries already set by plugin (e.g. 'output')
-			if !exists {
-				ret[k] = v
-			}
-		}
+	if lastNL == -1 {
```

Comment:
by returning early you're changing the code paths: exitStatus error wont be triggered here?

Reply:
`exitStatus` is already in metadata but you right, it's not consistent /w the previous version.
Fixed.